core/lib/spree/testing_support/factories/promotion_factory.rb

```diff
@@ -30,12 +30,14 @@
     trait :with_line_item_adjustment do
       transient do
         adjustment_rate { 10 }
+        calculator_class { Spree::Calculator::FlatRate }
+        promotion_action_class { Spree::Promotion::Actions::CreateItemAdjustments }
       end
 
       after(:create) do |promotion, evaluator|
-        calculator = Spree::Calculator::FlatRate.new
+        calculator = evaluator.calculator_class.new
         calculator.preferred_amount = evaluator.adjustment_rate
-        Spree::Promotion::Actions::CreateItemAdjustments.create!(calculator: calculator, promotion: promotion)
+        evaluator.promotion_action_class.create!(calculator: calculator, promotion: promotion)
```

Comment:
What about doing the same on the other traits in the file?

Reply:
Nice idea, let me know what you think of the new commits @kennyadsl!